lib/hsa/CMakeLists.txt

```diff
@@ -1,9 +1,13 @@
+include(FindPkgConfig)
+
 ####################
 # HCC runtime (HSA implementation)
 ####################
 if (HAS_ROCM EQUAL 1)
 add_mcwamp_library_hsa(mcwamp_hsa mcwamp_hsa.cpp unpinned_copy_engine.cpp)
-target_link_libraries(mcwamp_hsa  unwind)
+pkg_check_modules(LIBUNWIND REQUIRED libunwind)
+# TODO: we should probably add required CFLAGS and LDFLAGS from pkgconfig
+target_link_libraries(mcwamp_hsa ${LIBUNWIND_LIBRARIES})
```

Comment:
> The downstream libraries use the imported targets. The advantage is the paths to the library are not hard coded so it allows everything to be relocatable.

OK, thanks. I thought this was handled via hcc-config ldflags, but adding it is not a problem. the bigger problem is:
> This is to be used after installation, so users can do find_package(hcc) and then the user can use the hcc cmake targets(like hccrt or hc_am) directly in their project.

So we can't use it for hcc configure checks. Sounds like the pkgconfig libunwind check needs to be in two places. One for hcc and in hcc-config.cmake.in for downstream users. What would be the right place for the former? Can I leave the patch as is, and just add another check to hcc-config.cmake.in?